h2o-algos/src/test/java/hex/pca/PCATest.java

```diff
@@ -105,6 +103,207 @@
     }
   }
 
+  /*
+This unit test uses the prostate datasets with NAs, calculate the eigenvectors/values with original PCA implementation.
+Next, it calculates the eigenvectors/values using PCA with wide dataset flag set to true.  Then, we
+compare the eigenvalues/vectors from both methods and they should agree.  Dataset contains numerical and
+categorical columns.  As soon as we introduced categorical columns,  disagreement between original PCA
+and wide data set PCA increases.  However, both methods achieve same accuracy when we project the projected data back
+to the original dataset.  Hence, it is okay in this case for them not to agree closely.
+ */
+  @Test public void testWideDataSetsWithNAs() throws InterruptedException, ExecutionException {
+    PCAModel modelN = null;     // store PCA models generated with original implementation
+    PCAModel modelW = null;     // store PCA models generated with wideDataSet set to true
+    Frame train = null, scoreN = null, scoreW = null;
+    try {
+      train = parse_test_file(Key.make("prostate_cat.hex"), "smalldata/prostate/prostate_cat.csv");
+      train.vec(0).setNA(0);
+      train.vec(3).setNA(10);
+      train.vec(5).setNA(100);
+      PCAModel.PCAParameters parms = new PCAModel.PCAParameters();
+      parms._train = train._key;
+      parms._k = 7;
+      parms._transform = DataInfo.TransformType.STANDARDIZE;
+      parms._use_all_factor_levels = true;
+      parms._pca_method = PCAParameters.Method.GramSVD;
+      parms._impute_missing=false;
+      parms._seed = 12345;
+
+      PCA pcaParms = new PCA(parms);
+      modelN = pcaParms.trainModel().get(); // get normal data
+      scoreN = modelN.score(train);
+
+      PCA pcaParmsW = new PCA(parms);
+      pcaParmsW.setWideDataset(true);  // force to treat dataset as wide even though it is not.
+      modelW = pcaParmsW.trainModel().get();
+      scoreW = modelW.score(train);
+
+      // check to make sure eigenvalues and eigenvectors are the same
+      // compare eigenvectors and eigenvalues generated by original PCA and wide dataset PCA.
+      TestUtil.checkStddev(modelW._output._std_deviation, modelN._output._std_deviation, 1.5);
+      boolean[] flippedEig = TestUtil.checkEigvec( modelW._output._eigenvectors, modelN._output._eigenvectors, 2);
```

Comment:
Good point here.  I will do the following instead:
1. Implement the code in octave and obtain the eigenvalues, eigenvectors;
2. Compare the eigenvalues, eigenvectors from Octave with our H2O implementation.